src/components/form/index.js

```diff
@@ -0,0 +1,95 @@
+import React, { useState, useEffect } from 'react'
+import Card from '@material-ui/core/Card'
+import CardContent from '@material-ui/core/CardContent'
+import Typography from '@material-ui/core/Typography'
+import { makeStyles } from '@material-ui/core/styles'
+import Button from '@material-ui/core/Button'
+import Box from '@material-ui/core/Box'
+
+import Input from '../input'
+
+const useStyles = makeStyles((theme) => ({
+    root: {
+        '& .MuiCardContent-root': {
+            paddingLeft: 0,
+        },
+        marginBottom: '10px',
+        paddingLeft: theme.spacing(2),
+        paddingRight: theme.spacing(2),
+        paddingBottom: theme.spacing(2),
+    },
+    title: {
+        fontSize: 28,
+    },
+}))
+
+const patterns = {
+    discord: /^https:\/\/discord\.com\/api\/webhooks\//,
+    ddb: /^https:\/\/www\.dndbeyond\.com\/profile\/(\w|\d)+\/characters\/\d+$/,
+}
+
+const Form = ({ character = {}, index, onSubmit, handleRemove }) => {
+    const classes = useStyles()
+    const [name, setName] = useState('')
+    const [ddbUrl, setDdbUrl] = useState(character.ddbUrl)
+    const [discordUrl, setDiscordUrl] = useState(character.discordUrl)
+    const handleSubmit = (e) => {
+        e.preventDefault()
+        onSubmit({
+            index,
+            character: {
+                ddbUrl,
+                discordUrl,
+            },
+        })
+    }
+
+    useEffect(() => {
+        async function fetchCharacterName() {
+            const htmlString = await fetch(ddbUrl).then((r) => r.text())
+            const htmlDoc = new DOMParser().parseFromString(
+                htmlString,
+                'text/html'
+            )
+            const titleText = htmlDoc.querySelector('title').innerText
+
+            setName(titleText.replace(' - D&D Beyond', ''))
```

Comment:
That's definitely an option and would `replace` unnecessary. My reasoning for preferring the title tag:

Class names aren't really indicative of dom structure, they could easily just throw a wrapper div or span around the returned element and it would break. 

The title tag, IMO, is less likely to change. I think from a UX perspective they would want to always have your character's name in the title tag, specifically in the first half so that it's visible to the player with multiple tabs open.

That said, I get your point. The proposed change would break if they had something other than ` - D&D Beyond` in the title tag.

Anyway, I'm happy to change it if you'd prefer it!

Reply:
You're right, the more I think about it, the title is probably less likely to change. I'm just not looking forward to the day when the DnD Beyond devs do a refactor of the website and we have to re-write this whole thing 😭